app/models/stat.rb

```diff
@@ -1,73 +1,163 @@
 class Stat
   include ActiveModel::Model
 
-  attr_accessor :applicants, :invitations, :agents, :organisations, :rdvs
+  attr_accessor :applicants, :invitations, :agents, :organisations, :rdvs, :rdv_contexts
 
+  # We don't include in the scope the organisations who don't invite the applicants
   def relevant_organisations
     @relevant_organisations ||= organisations
                                 .includes(:applicants, :rdvs)
                                 .joins(:configurations)
                                 .where(configurations: { notify_applicant: false })
   end
 
+  # We filter the applicants by organisations and retrieve deleted or archived applicants
   def relevant_applicants
     @relevant_applicants ||= applicants
-                             .includes(:rdvs)
                              .joins(:organisations)
                              .where(organisations: relevant_organisations)
                              .where.not(status: %w[deleted])
-                             .where(is_archived: false)
+                             .archived(false)
+                             .distinct
   end
 
+  # We don't include in the stats the agents working for rdv-insertion
+  def relevant_agents
+    agents.where.not("agents.email LIKE ?", "%beta.gouv.fr").uniq
+  end
+
+  # We filter the rdvs to keep the rdvs of the applicants in the scope
   def relevant_rdvs
-    @relevant_rdvs ||= rdvs.joins(:applicants).where(applicants: relevant_applicants)
+    @relevant_rdvs ||= rdvs.joins(:applicants).where(applicants: relevant_applicants).distinct
   end
 
-  def percentage_of_applicants_oriented_in_time
-    (applicants_oriented_in_less_than_30_days.length / (applicants_orientable_in_time.length.nonzero? || 1).to_f) * 100
+  # For the % of no show, the average rdv delay and the rate of applicants oriented in less than 30 days
+  # we only consider the rdvs in the "rsa_orientation" contexts, because the other rdvs are not always
+  # correctly informed by the organisations/ or are taken in the past (which mess up with the delays)
+  def orientation_rdvs
+    @orientation_rdvs ||= relevant_rdvs.joins(:rdv_contexts)
+                                       .where(rdv_contexts: { context: %w[rsa_orientation] })
   end
 
-  def applicants_oriented_in_less_than_30_days
-    relevant_applicants.to_a.select do |applicant|
-      applicant.rdv_seen? && applicant.orientation_delay_in_days && applicant.orientation_delay_in_days < 30
-    end
+  def orientation_rdvs_by_month
+    @orientation_rdvs_by_month ||= orientation_rdvs.group_by { |m| m.created_at.beginning_of_month }
+  end
+
+  def relevant_rdv_contexts
+    @relevant_rdv_contexts ||= rdv_contexts.where(applicant_id: relevant_applicants.pluck(:id))
+                                           .joins(:rdvs).where.not(rdvs: { id: nil })
+                                           .joins(:invitations).where.not(invitations: { id: nil })
```

Comment:
Tu peux utiliser la nouvelle méthode de rails [`associated`](https://www.bigbinary.com/blog/rails-6-1-adds-where-associated-to-check-association-presence) ici
```suggestion
                                           .where.associated(:rdvs)
                                           .where.associated(:invitations)
```

Reply:
🤯🤯🤯
Tellement bien cette nouvelle méthode, en fait je suis étonné qu'elle n'ait pas été faite plus tôt